src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs

```diff
@@ -36,15 +34,19 @@ public KestrelServer(IFeatureCollection features, KestrelServerOptions options,
                 throw new ArgumentNullException(nameof(applicationLifetime));
             }
 
-            if (logger == null)
+            if (loggerFactory == null)
             {
-                throw new ArgumentNullException(nameof(logger));
+                throw new ArgumentNullException(nameof(loggerFactory));
             }
 
+            Options = options.Value ?? new KestrelServerOptions();
             _applicationLifetime = applicationLifetime;
-            _logger = logger;
-            Features = features;
-            Options = options;
+            _logger = loggerFactory.CreateLogger("Microsoft.AspNetCore.Server.Kestrel");
+            Features = new FeatureCollection();
+            var componentFactory = new HttpComponentFactory(Options);
+            Features.Set<IHttpComponentFactory>(componentFactory);
+            _serverAddresses = new ServerAddressesFeature();
+            Features.Set<IServerAddressesFeature>(_serverAddresses);
```

Comment:
I thought Hosting was going to set the `IServerAddressesFeature` and the servers could merely replace it if they want to?

@davidfowl